src/chat.rs

```diff
@@ -894,8 +893,16 @@ impl ChatId {
                     .await?
                     .context("no file stored in params")?;
                 msg.param.set(Param::File, blob.as_name());
-                if blob.suffix() == Some(WEBXDC_SUFFIX) {
-                    msg.viewtype = Viewtype::Webxdc;
+                if msg.viewtype == Viewtype::File {
+                    if let Some((better_type, _)) =
+                        message::guess_msgtype_from_suffix(&blob.to_abs_path())
+                            .filter(|&(vt, _)| vt == Viewtype::Webxdc || vt == Viewtype::Vcard)
```

Comment:
Suggestion, non-blocking: Could this get a comment like "If the attached file is a VCard or a Webxdc, automatically convert it to a VCard/Webxdc. We do not do this for other files, because we recommend that users can send images as a “file” to preserve the original quality (usually, we compress some files like images)."?

Reply:
Added the comment, only slightly changed it